devices/MS5611/Ms5611.cs

```diff
@@ -0,0 +1,201 @@
+        /// </param>
+        /// <returns>
+        ///  Pressure
+        /// </returns>
+        public Pressure ReadSeaLevelPressure(Length altitude) => WeatherHelper.CalculateSeaLevelPressure(ReadPressure(), altitude, ReadTemperature());
+
+        /// <summary>
+        ///  Reads raw temperature value from the sensor
+        /// </summary>
+        /// <param name="compensation" >
+        ///  Flag indicating if compensation calculation for low temperatures (lower than 20 degrees C) should be done
+        /// </param>
+        /// <returns>
+        ///  Temperature in Celsius
+        /// </returns>
+        public double ReadTrueTemperature(bool compensation = false)
+
+        {
+            long rawTemp = ReadRawData((byte)((byte)CommandAddresses.SamplingRateTemperature + _sampling));
+            long diffActualAndReference = rawTemp - _calibrationData.ReferenceTemperature * 256;
+            double actualTemperature = 2000 + (double)diffActualAndReference * _calibrationData.TemperatureCoefficientOfTheTemperature / 8388608;
+            double compensationValue = 0;
+            if (compensation)
+            {
+                if (actualTemperature < 2000)
+                {
+                    compensationValue = diffActualAndReference * diffActualAndReference / (2 << 30);
+                }
+            }
+            var temp = actualTemperature - compensationValue;
+            return (double)temp / 100;
+        }
+
+        internal int ReadRegister(int numberOfByte)
+        {
+            _i2cDevice.WriteByte((byte)numberOfByte);
+            var readData = new byte[2];
+            _i2cDevice.Read(new SpanByte(readData));
+            int rawData = readData[0] << 8 | readData[1];
+            return rawData;
+        }
+
+        internal long ReadRawData(byte address)
+        {
+            _i2cDevice.WriteByte(address);
+            Thread.Sleep(_delayForSampling);
+            _i2cDevice.WriteByte((byte)CommandAddresses.AdcRead);
+            var readData = new byte[3];
+            _i2cDevice.Read(new SpanByte(readData));
+            long rawData = readData[0] << 16 | readData[1] << 8 | readData[2];
+            return rawData;
+        }
+
+        /// <summary>
+        ///  Reads raw pressure value from the sensor
+        /// </summary>
+        /// <param name="compensation" >
+        ///  Flag indicating if compensation calculation for low temperatures (lower than 20 degrees C) should be done
+        /// </param>
+        /// <returns>
+        ///  Pressure in millibars
+        /// </returns>
+        public double ReadTruePressure(bool compensation = false)
+        {
+            long rawPressure = ReadRawData((byte)((byte)CommandAddresses.SamplingRatePressure + _sampling));
+            long rawTemp = ReadRawData((byte)((byte)CommandAddresses.SamplingRateTemperature + _sampling));
+            long diffActualAndReferenceTemperature = (long)(rawTemp - _calibrationData.ReferenceTemperature * 256);
```

Comment:
and please refer to the doc for the calculation, a lot of values that may not be so easy to maintain